src/sentry/workflow_engine/migration_helpers/rule.py

```diff
@@ -205,6 +203,39 @@ def update_dcg(
     return dcg
 
 
+def delete_migrated_issue_alert(rule: Rule):
+    try:
+        alert_rule_workflow = AlertRuleWorkflow.objects.get(rule=rule)
+    except AlertRuleWorkflow.DoesNotExist:
+        logger.exception("AlertRuleWorkflow does not exist", extra={"rule_id": rule.id})
+        return
+
+    workflow: Workflow = alert_rule_workflow.workflow
+
+    try:
+        if_dcg = WorkflowDataConditionGroup.objects.get(workflow=workflow).condition_group
+        DataCondition.objects.filter(condition_group=if_dcg).delete()
```

Comment:
last nit: maybe we should filter here in case there is more than 1 IF DCG for whatever reason

```suggestion
        if_dcg = WorkflowDataConditionGroup.objects.get(workflow=workflow).condition_group
        if_dcg.conditions.all().delete()
```